test/functional/tests/cache_ops/test_core_remove.py

```diff
@@ -0,0 +1,100 @@
+
+mount_point = "/mnt/cas"
+cores_amount = 3
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_remove_core_when_other_mounted_auto_numeration():
+    """
+        title: |
+          Test for removing one core from the cache when the other core is mounted.
+          Cores are numerated automatically.
+        description: |
+          Test of the ability to remove the unmounted core from the cache when the other core
+          is mounted and its ID starts with a different digit.
+        pass_criteria:
+          - No system crash.
+          - Removing unmounted core finished with success.
+    """
+    with TestRun.step("Prepare devices."):
+        cache_device = TestRun.disks['cache']
+        cache_device.create_partitions([Size(50, Unit.MebiByte)])
+        cache_part = cache_device.partitions[0]
+        core_device = TestRun.disks['core']
+        core_device.create_partitions([Size(200, Unit.MebiByte)] * 2)
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, force=True)
+
+    with TestRun.step("Add cores to cache and mount it."):
+        cores = []
+        for i, part in enumerate(core_device.partitions):
+            part.create_filesystem(Filesystem.xfs)
+            cores.append(cache.add_core(part))
+            if i > 0:  # leaving the 1st core unmounted
+                cores[i].mount(f"{mount_point}{cache.cache_id}-{cores[i].core_id}")
+
+    with TestRun.step("Remove the unmounted core."):
+        try:
+            cache.remove_core(cores[0].core_id)
+        except CmdException as exc:
+            TestRun.fail(f"Cannot remove the unmounted core.\n{exc}")
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_remove_core_when_other_mounted_custom_numeration():
+    """
+        title: |
+          Test for removing one core from the cache when the other core is mounted.
+          Cores have custom numeration, starting with the same digit.
+        description: |
+          Test of the ability to remove the unmounted core from the cache when the other core
+          is mounted and its ID starts with the same digit.
+        pass_criteria:
+          - No system crash.
+          - Removing unmounted core finished with success.
+    """
+    with TestRun.step("Prepare devices."):
+        cache_device = TestRun.disks['cache']
+        cache_device.create_partitions([Size(50, Unit.MebiByte)])
+        cache_part = cache_device.partitions[0]
+        core_device = TestRun.disks['core']
+        core_device.create_partitions([Size(200, Unit.MebiByte)] * cores_amount)
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_part, force=True)
+
+    with TestRun.step("Add cores to cache and mount it."):
+        cores = []
```

Comment:
List seems redundant, you can keep reference to not mounted core in regular variable.